test_unstructured_ingest/check-diff-expected-output.sh

```diff
@@ -14,12 +14,30 @@ set +e
 
 SCRIPT_DIR=$(dirname "$(realpath "$0")")
 OVERWRITE_FIXTURES=${OVERWRITE_FIXTURES:-false}
+TMP_DIRECTORY_CLEANUP=${TMP_DIRECTORY_CLEANUP:-true}
 OUTPUT_FOLDER_NAME=$1
 OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
 OUTPUT_DIR_TEXT=$SCRIPT_DIR/text-output/$OUTPUT_FOLDER_NAME
 EXPECTED_OUTPUT_DIR=$SCRIPT_DIR/expected-structured-output/$OUTPUT_FOLDER_NAME
 EXPECTED_OUTPUT_DIR_TEXT=$SCRIPT_DIR/expected-text-output/$OUTPUT_FOLDER_NAME
 
+function cleanup() {
+  if [ "$TMP_DIRECTORY_CLEANUP" == "true" ]; then
+    if [ -d "$EXPECTED_OUTPUT_DIR_TEXT" ]; then
+      echo "cleaning up tmp directory: $EXPECTED_OUTPUT_DIR_TEXT"
+      rm -rf "$EXPECTED_OUTPUT_DIR_TEXT"
+    fi
+    if [ -d "$OUTPUT_DIR_TEXT" ]; then
+      echo "cleaning up tmp directory: $OUTPUT_DIR_TEXT"
+      rm -rf "$OUTPUT_DIR_TEXT"
+    fi
+  else
+    echo "skipping tmp directory cleanup"
+  fi
+}
+
+trap cleanup EXIT
+
```

Comment:
if we're calling cleanup in individual scripts, do we need to do it here?
cleanup here feels like an unexpected consequence when calling check-diff

Reply:
this cleanup is specific to the text files generated within this script.